src/main.py

```diff
@@ -1121,6 +1171,7 @@ def _return_to_cluster(self, new_primary, role, is_dead=False):
             self.checks['primary_switch'] = 1
         logging.debug("primary_switch checks is %d", self.checks['primary_switch'])
 
+        self._acquire_replication_source_lock(new_primary)
```

Comment:
See comment on https://github.com/yandex/pgconsul/pull/30/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R657
It looks like a repeated call during single iteration.

Reply:
https://github.com/yandex/pgconsul/pull/30#discussion_r1696408494